client/state/data-layer/http-data.ts

```diff
@@ -317,7 +321,7 @@ export const waitForData = < T extends Query >(
 		};
 
 		if ( timeout ) {
-			timer = setTimeout( () => {
+			timer = ( setTimeout as typeof window.setTimeout )( () => {
```

Comment:
how do you feel about this? seems like some major hoops just to get typing right…

Reply:
I'd prefer not to jump through hoops, do you know why b36ec842523b9a8b1a10988c6deb26117fec76cf made this unnecessary?

I also wonder if separate client and server apps that could know more about their runtime could do better with this kind of ambiguity.